control-plane/config/eventing-kafka-source/200-controller/500-controller.yaml

```diff
@@ -0,0 +1,118 @@
+---
+
+# Copyright 2022 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kafka-controller
+  namespace: knative-eventing
+  labels:
+    app: kafka-controller
+    kafka.eventing.knative.dev/release: devel
+spec:
+  selector:
+    matchLabels:
+      app: kafka-controller
+  template:
+    metadata:
+      name: kafka-controller
+      labels:
+        app: kafka-controller
+        kafka.eventing.knative.dev/release: devel
+    spec:
+      securityContext:
+        runAsNonRoot: true
+      serviceAccountName: kafka-controller
+
+      # To avoid node becoming SPOF, spread our replicas to different nodes.
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    app: kafka-controller
+                topologyKey: kubernetes.io/hostname
+              weight: 100
```

Comment:
Should we add zone-based topology constraints as in https://github.com/knative-sandbox/eventing-kafka-broker/pull/2092?

Reply:
Hmm even spread across zones doesn't make sense here for this use-case... since we'll only have one replica for the controller. So no need?

Reply:
In theory, for control plane HA, we might want to have multiple controllers running and spread across zones and nodes.